framework/tests/src/tests/signatureinference/SignatureInferenceTestAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,162 @@
+        implicitsTreeAnnotator.addTreeKind(
+                com.sun.source.tree.Tree.Kind.INT_LITERAL, bottom);
+
+        return new ListTreeAnnotator(new PropagationTreeAnnotator(this),
+                implicitsTreeAnnotator);
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
+        return new SignatureTestQualifierHierarchy(factory);
+    }
+
+    /**
+     * Using a MultiGraphQualifierHierarchy to enable tests with Annotations
+     * that contain fields. @see SiblingWithFields.
+     * @author pbsf
+     *
+     */
+    protected class SignatureTestQualifierHierarchy extends
+                MultiGraphQualifierHierarchy {
+
+        public SignatureTestQualifierHierarchy(MultiGraphFactory f) {
+            super(f);
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1,
+                AnnotationMirror a2) {
+            if ((AnnotationUtils.areSameByClass(a1, Sibling1.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling2.class)) 
+                    || (AnnotationUtils.areSameByClass(a1, Sibling2.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling1.class))
+                    || (AnnotationUtils.areSameByClass(a1, Sibling1.class)
+                    && AnnotationUtils.areSameByClass(a2, SiblingWithFields.class))
+                    || (AnnotationUtils.areSameByClass(a1, SiblingWithFields.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling2.class))
+                    || (AnnotationUtils.areSameByClass(a1, Sibling2.class)
+                    && AnnotationUtils.areSameByClass(a2, SiblingWithFields.class))) {
+                return PARENT;
+            }
+            return super.leastUpperBound(a1, a2);
+        }
+        @Override
+        public boolean isSubtype(AnnotationMirror sub, AnnotationMirror sup) {
+            if (AnnotationUtils.areSame(sub, sup)
+                    || AnnotationUtils.areSameByClass(sup, UnknownClass.class)
+                    || AnnotationUtils.areSameByClass(sub, SignatureInferenceBottom.class)
+                    || AnnotationUtils.areSameByClass(sup, Top.class)) {
+                return true;
+            }
+
+            if (AnnotationUtils.areSameByClass(sub, UnknownClass.class)
+                    || AnnotationUtils
+                            .areSameByClass(sup, SignatureInferenceBottom.class)) {
+                return false;
+            }
+
+            if (AnnotationUtils.areSameByClass(sub, Top.class)) {
+                return false;
+            }
+
+            if ((AnnotationUtils.areSameByClass(sub, Sibling1.class)
+                    || AnnotationUtils.areSameByClass(sub, Sibling2.class)
+                    || AnnotationUtils.areSameByClass(sub, SiblingWithFields.class))
+                    && AnnotationUtils.areSameByClass(sup, Parent.class)) {
```

Comment:
This indentation is misleading too.  The "||" should be aligned under the "A" on the first line, but the "&&" should be aligned under the "(" that precedes the "A".